### PR TITLE
[osmanip] Add new port

### PR DESCRIPTION
**Pull request by JustWhit3**
**Describe the pull request**

- #### What does your PR fix?
  This PR adds port to [osmanip](https://github.com/JustWhit3/osmanip).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Linux, MacOS, Windows. Yes.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/osmanip/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "name": "osmanip",
+  "version": "4.5.0",
+  "description": "A cross-platform library for output stream manipulation using ANSI escape sequences.",
+  "homepage": "https://github.com/JustWhit3/osmanip",
+  "license": "MIT",
+  "dependencies": [
+    {
+      "name": "arsenalgear",
+      "host": true
+    },
```
Comment (dg0yt) [inline ports/osmanip/vcpkg.json:11]: Do you really need this for the host? I guess you want this for the target, i.e. just plain `"arsenalgear"`.

Comment (JustWhit3) [reply ports/osmanip/vcpkg.json]: I sincerly don't know how to treat dependencies in this file, so I put also the host but probably it was wrong. Just corrected.

Comment (dg0yt) [reply ports/osmanip/vcpkg.json]: The change seems to have a positive effect :-)

`"host": true` is for scripts or executables which are run during the built, not for code which becomes linked to the target artifacts. Full reference for dependency fields:
https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#dependency

Comment (JustWhit3) [reply ports/osmanip/vcpkg.json]: Thanks!

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Feature `arsenalgear` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

https://github.com/JustWhit3/osmanip/blob/4c810a579eba784af0e86c1bec10cb33832a8e63/CMakeLists.txt#L47-L48 may be changed to:
````
    find_package(arsenalgear CONFIG REQUIRED)
    target_link_libraries(main PRIVATE arsenalgear::arsenalgear)
````
because I tested usage failed:
````
 CMake Error at F:/Feature-test/osmanip/vcpkg/scripts/buildsystems/vcpkg.cmake:585 (_add_executable):
1> [CMake]   Target "main" links to target "arsenalgear::arsenalgear" but the target was
1> [CMake]   not found.  Perhaps a find_package() call is missing for an IMPORTED
1> [CMake]   target, or an ALIAS target is missing?
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeProject42/CMakeLists.txt:14 (add_executable)
1> [CMake] -- Generating done
````
when I add `find_package(arsenalgear CONFIG REQUIRED)` it worked.

**Comment by dg0yt** [comment]
The suggested osmanip change is good (with regard to the `CONFIG`, `REQUIRED` and `PRIVATE` keywords), but what is really missing is
~~~cmake
include(CMakeFindDependencyMacro)
find_dependency(arsenalgear CONFIG)
~~~
early in osmanip's  `cmake/Config.cmake.in`.

**Comment by JustWhit3** [comment]
Hello,

@dg0yt I added your updates, now it should be fixed.

@JonLiu1993 I can't add your updates since if I do this I get an error:

```
CMake Error at CMakeLists.txt:47 (find_package):
  Could not find a package configuration file provided by "arsenalgear" with
  any of the following names:

    arsenalgearConfig.cmake
    arsenalgear-config.cmake

  Add the installation prefix of "arsenalgear" to CMAKE_PREFIX_PATH or set
  "arsenalgear_DIR" to a directory containing one of the above files.  If
  "arsenalgear" provides a separate development package or SDK, be sure it
  has been installed.
```

**Comment by JustWhit3** [comment]
@JonLiu1993 I put that line of code exactly [here](https://github.com/JustWhit3/osmanip/blob/4c810a579eba784af0e86c1bec10cb33832a8e63/CMakeLists.txt#L47-L48) into the `osmanip` CMakeLists.txt file. However, is it important to have `REQUIRED` in order to let it work? It seems all fine without it.

**Comment by JonLiu1993** [comment]
> @JonLiu1993 I put that line of code exactly [here](https://github.com/JustWhit3/osmanip/blob/4c810a579eba784af0e86c1bec10cb33832a8e63/CMakeLists.txt#L47-L48) into the `osmanip` CMakeLists.txt file. However, is it important to have `REQUIRED` in order to let it work? It seems all fine without it.

You have added to cmake/config.cmake.in
````
Include (CmakefindDependencyMacro)
Find_Dependency (ARSENALGEAR Config)
````
It will automatically modify the files in CMAKELISTS. What you have to do now is to make a patch S manually modify the usage in CMAKELISTS to ensure that users can use it for the time being
````
Find_package (ARSENALGEAR)
target_link_libraries (OSMANIP PUBLIC ARSENALGEAR :: ARSENALGEAR)
````
After the next stable version is released, this patch is no longer needed

**Comment by JustWhit3** [comment]
> > @JonLiu1993 I put that line of code exactly [here](https://github.com/JustWhit3/osmanip/blob/4c810a579eba784af0e86c1bec10cb33832a8e63/CMakeLists.txt#L47-L48) into the `osmanip` CMakeLists.txt file. However, is it important to have `REQUIRED` in order to let it work? It seems all fine without it.
> 
> You have added to cmake/config.cmake.in
> 
> ```
> Include (CmakefindDependencyMacro)
> Find_Dependency (ARSENALGEAR Config)
> ```
> 
> It will automatically modify the files in CMAKELISTS. What you have to do now is to make a patch S manually modify the usage in CMAKELISTS to ensure that users can use it for the time being
> 
> ```
> Find_package (ARSENALGEAR)
> target_link_libraries (OSMANIP PUBLIC ARSENALGEAR :: ARSENALGEAR)
> ```
> 
> After the next stable version is released, this patch is no longer needed

Updated. Now it should be correct.

**Comment by JonLiu1993** [comment]
@JustWhit3, Sorry, maybe my statement is not clear enough, the only thing you need to do now is to make a patch, the patch content is to https://github.com/JustWhit3/osmanip/blob/4c810a579eba784af0e86c1bec10cb33832a8e63/CMakeLists.txt#L47-L48
````
    find_package(arsenalgear)
    target_link_libraries( osmanip PUBLIC arsenalgear::arsenalgear )
````
change into
````
     find_package(arsenalgear CONFIG REQUIRED)
     target_link_libraries(main PRIVATE arsenalgear::arsenalgear)
````

**Comment by dg0yt** [comment]
AFAICS upstream source has capabilities to fetch arsenalgear. This may hide  the lack to properly use vcpkg's arsenalgear (`REQUIRED`). This structure should be improved.

https://github.com/JustWhit3/osmanip/blob/fd70ee63c32f2b4d05f9092cb3b938af6928d867/CMakeLists.txt#L26-L28

https://github.com/JustWhit3/osmanip/blob/fd70ee63c32f2b4d05f9092cb3b938af6928d867/deps/arsenalgear/CMakeLists.txt#L18-L31

**Comment by JustWhit3** [comment]
@JonLiu1993 Ok I added a new patch and now it should be correct.

@dg0yt So is it right what I did? I sincerly didn't completely understand your statement, sorry.

File: ports/osmanip/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Github config
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO JustWhit3/osmanip
+    REF "v${VERSION}"
+    SHA512 7cd03974e3f78e2531e561833830d66bc2e0dc115aed48f69c6cdf1c4b8a2bd18b05444003f816f670f41aeb08dc899f064de9f4c5b369118f1892e9dedf1204
```
Comment (JonLiu1993) [inline ports/osmanip/portfile.cmake:6]:
```suggestion
    SHA512 cbbae779435bec3995756e1dbc8c283868dd923453054177dfa73bc00ac3510488467e535a1c4b14726f02d198f298eaeb884c1716dcde07e9aec06d02b22d32
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/osmanip/portfile.cmake (hunk 2 above)
Comment (JustWhit3) [reply ports/osmanip/portfile.cmake]: Applied.

**Comment by JonLiu1993** [comment]
Usage tested successfully by `osmanip:x64-windows`:
````
    # this is heuristically generated, and may not be correct
    find_package(osmanip CONFIG REQUIRED)
    target_link_libraries(main PRIVATE osmanip::osmanip)
````

**Review by JonLiu1993 — DISMISSED** [review]

**Pull request merged**